Add keyboard support for slider widget

pkaczmarczy opened the pull request:
[Issue] https://github.com/Samsung/TAU/issues/158, https://github.com/Samsung/TAU/issues/157
[Problem] Slider should be able to implement it's own keyboard
          handling and switch to this control after beeing
          selected by KeyboardManager. "Change" event was not
          triggered after value change.
[Solution] Add possibility to delegate keyboard
           event handling to widget instance if needed.

Signed-off-by: Pawel Kaczmarczyk <p.kaczmarczy@samsung.com>

pkaczmarczy committed:
Add keyboard support for slider widget

[Issue] https://github.com/Samsung/TAU/issues/158, https://github.com/Samsung/TAU/issues/157
[Problem] Slider should be able to implement it's own keyboard
          handling and switch to this control after beeing
          selected by KeyboardManager. "Change" event was not
          triggered after value change.
[Solution] Add possibility to delegate keyboard
           event handling to widget instance if needed.

Signed-off-by: Pawel Kaczmarczyk <p.kaczmarczy@samsung.com>

TomaszLukawskiSam commented on src/js/core/widget/core/Slider.js:
```diff
@@ -715,88 +723,138 @@
+				events.trigger(self.element, "input");
 			};
 
-			// prototype._lockKeyboard = function () {
```
I cannot verify this change during this review.

TomaszLukawskiSam replied:
I wonder if we should remove this code

pkaczmarczy replied:
This code was used to lock keyboard support inside selected slider.
For now this is mechanism is handled in different way. I have added to base keyboard support key event delegation if any of the widgets after "actionEnter" will capture events.
Also disabling event delegation is now handled by BaseKeyboardSupport not by widget itself.

I think we can safely remove this code.

pkaczmarczy committed:
Add focus() and blur() on Slider input element

[Issue] https://github.com/Samsung/TAU/issues/162
[Problem] Page content was not scrolled
          to show currently focused slider
[Solution] Add focus() calls for input element
           and allow browser to scroll content

Signed-off-by: Pawel Kaczmarczyk <p.kaczmarczy@samsung.com>

TomaszLukawskiSam commented on src/js/core/widget/core/Slider.js:
```diff
@@ -611,14 +623,10 @@
+						case KEYBOARD_SUPPORT_EVENTS.taushortkeypress:
+							self._onTauShortKeyPress(event);
+							break;
+						case KEYBOARD_SUPPORT_EVENTS.taulongkeypress:
```
This code nothing does.

pkaczmarczy replied:
I have removed this code.

pkaczmarczy replied:
This lines have been removed.

TomaszLukawskiSam commented on src/js/core/widget/core/Slider.js:
```diff
@@ -715,88 +723,138 @@
 				var self = this;
 
 				self._setValue(self._value - (parseFloat(self.element.step) || 1));
+				events.trigger(self.element, "input");
```
I'm afraid that this event may cause looping in app eg. smart-things. This should be checked.

pkaczmarczy committed:
Prevent locking focus by elements that can capture keyboard

[Issue] N/A
[Problem] In case if history.back() was called before widget
          released keyboard capturing, focus would have been
          locked on element from different page. This could
          cause inability to navigate on app with keyboard.
[Solution] Clean focus before showing different page

Signed-off-by: Pawel Kaczmarczyk <p.kaczmarczy@samsung.com>

pkaczmarczy commented:
Thank you for your review :)
I have updated the PR, could please do a review once again?

lslachciak commented:
Because currently we are not considering this feature, could you close this PR. We will reopen it when needed.

lslachciak commented:
After discussing with @pkaczmarczy I'm closing this issue.
We will reopen and improve when this feature required
